pkg/admission/zone_tracker_test.go

```diff
@@ -286,3 +295,150 @@ func TestLastDownscaledNonExistentZone(t *testing.T) {
+				},
+				Resource: metav1.GroupVersionResource{
+					Group:    "apps",
+					Version:  "v1",
+					Resource: "statefulsets",
+				},
+				Name:      stsName,
+				Namespace: namespace,
+				Object: runtime.RawExtension{
+					Raw: rawObject,
+				},
+				OldObject: runtime.RawExtension{
+					Raw: oldRawObject,
+				},
+				DryRun: &dryRun,
+			},
+		}
+	}
+
+	api := fake.NewSimpleClientset()
+
+	zt := newZoneTracker(api, namespace, "testconfigmap")
+
+	// block the ingester-zone-a downscale request for rollout group ingester
+	ingesterZoneAPrepDownscaleDone := make(chan struct{})
+	ingesterZoneADownscaleInitiated := atomic.Bool{}
+	go func() {
+		f := newFakeHttpClient(func(r *http.Request) (*http.Response, error) {
+			ingesterZoneADownscaleInitiated.Store(true)
+			<-ingesterZoneAPrepDownscaleDone
+			return &http.Response{
+				StatusCode: http.StatusOK,
+				Body:       io.NopCloser(bytes.NewBuffer([]byte(""))),
+			}, nil
+		})
+
+		ar := buildAdmissionRequest(rolloutGroupIngester, ingesterZoneA)
+		admissionResponse := zt.prepareDownscale(context.Background(), logger, ar, api, f)
+		require.True(t, admissionResponse.Allowed, admissionResponse.Result.Message)
+	}()
+
+	// wait for ingester-zone-a downscale request to get accepted before checking rejection of downscale requests for ingester group
+	require.Eventually(t, func() bool {
+		return ingesterZoneADownscaleInitiated.Load()
+	}, time.Second, time.Millisecond)
+
+	// ingester-zone-b downscale request for rollout group ingester should get rejected
+	ar := buildAdmissionRequest(rolloutGroupIngester, ingesterZoneB)
+	admissionResponse := zt.prepareDownscale(context.Background(), logger, ar, api, f)
+	require.False(t, admissionResponse.Allowed)
+	require.Equal(t, "downscale of statefulsets/ingester-zone-b in test from 5 to 2 replicas is not allowed because statefulset ingester-zone-a is already in process of updating replicas", admissionResponse.Result.Message)
+
+	// no zones should have been updated
+	require.NoError(t, zt.loadZones(context.Background(), nil))
+	require.Len(t, zt.zones, 0)
+
+	// while downscale request for group index-gateway should pass
+	ar = buildAdmissionRequest(rolloutGroupIndexGateway, indexGatewayZoneA)
+	admissionResponse = zt.prepareDownscale(context.Background(), logger, ar, api, f)
+	require.True(t, admissionResponse.Allowed)
+
+	// only index-gateway-zone-a should have been updated
+	require.NoError(t, zt.loadZones(context.Background(), nil))
+	require.Len(t, zt.zones, 1)
+	_, zoneUpdated := zt.zones[indexGatewayZoneA]
+	require.True(t, zoneUpdated)
+
+	// finishing the in progress downscaling request for rollout group ingester should let new requests to go through
+	close(ingesterZoneAPrepDownscaleDone)
+
+	require.Eventually(t, func() bool {
+		ar = buildAdmissionRequest(rolloutGroupIngester, ingesterZoneB)
+		return zt.prepareDownscale(context.Background(), logger, ar, api, f).Allowed
+	}, 5*time.Second, time.Millisecond)
```

Comment:
Running this every millisecond might be a little too aggressive - how about we run it a little less frequently?

```suggestion
	}, 5*time.Second, 50*time.Millisecond)
```

Reply:
makes sense. Thanks.